### PR TITLE
[Chat] fix eval: add in training evaluation

### PR DESCRIPTION
**Pull request by YeAnbang**
## 📌 Checklist before creating the PR

- [ ] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [ ] I have added relevant tags if possible for us to better distinguish different PRs
- [x] I have installed pre-commit: `pip install pre-commit && pre-commit install`

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.

## 💥 Checklist before requesting a review

- [ ] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [ ] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [x] I have performed a self-review of my code
- [x] I have added thorough tests.
- [x] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [x] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: applications/ColossalChat/coati/trainer/orpo.py [hunk 1]
```diff
@@ -283,14 +283,22 @@ def _eval(self, epoch: int):
                 logprob_actor_reject = calc_masked_log_probs(
                     actor_reject_logits, reject_input_ids, reject_loss_mask[:, 1:]
                 )
-
-                odds_ratio_loss, log_odds_ratio = self.odds_ratio_loss_fn(logprob_actor_chosen, logprob_actor_reject)
-
+                chosen_logits = actor_chosen_logits[:, :-1, :].contiguous().view(-1, actor_chosen_logits.size(-1))
+                label_chosen = chosen_input_ids[:, 1:].contiguous()
```
Comment (TongLi3701) [inline applications/ColossalChat/coati/trainer/orpo.py:287]: For labels, why do we need to shift manually? crossentropy loss should automatically shift and calculate loss?

**Review by TongLi3701 — COMMENTED** [review]
Thanks Anbang, some questions.

File: applications/ColossalChat/coati/trainer/orpo.py (hunk 1 above)
Comment (YeAnbang) [reply applications/ColossalChat/coati/trainer/orpo.py:287]: This is a bug, I will fix it, thanks

File: applications/ColossalChat/coati/dataset/loader.py [hunk 2]
```diff
@@ -28,6 +28,8 @@ def load_tokenized_dataset(
     Each instance of dataset is a dictionary with
     `{'input_ids': List[int], 'labels': List[int], sequence: str}` format.
     """
+    if not dataset_paths:
+        return None
```
[the remark this answers is not in the document]
Comment (YeAnbang) [reply applications/ColossalChat/coati/dataset/loader.py:32]: added warning to the training script to throw out a warning is evaluation dataset is not provided

**Review by TongLi3701 — APPROVED** [review]
Thanks Anbang.

Overall, looks good. 

Please have a look at the code refactor part to make sure the tests are passed.
